shared/js/site.js

```diff
@@ -52,10 +53,41 @@ class Score {
                 // remove period at end for lookup in pagesSeenOn
                 let tosdrData = tosdr[match[0]]
 
+                // tosdr message
+                // 1. If we have a defined tosdr class look up the message in constants
+                //    for the corresponding letter class
+                // 2. If there are both good and bad points -> 'mixed'
+                // 3. Else use the calculated tosdr score to determine the message
+                let message = constants.tosdrMessages.unknown
+                if (tosdrData.class) {
+                    message = constants.tosdrMessages[tosdrData.class]
+                } else if (tosdrData.match.good.length && tosdrData.match.bad.length) {
+                    message = constants.tosdrMessages.mixed
+                } else {
+                    if (tosdrData.score < 0) {
+                        message = constants.tosdrMessages.good  
+                    } else if (tosdrData.score === 0 && (tosdrData.match.good.length || tosdrData.match.bad.length)) {
+                        message = constants.tosdrMessages.mixed
+                    } else if (tosdrData.score > 0 ) {
+                        message = constants.tosdrMessages.bad
+                    }
+                }
+
+                let reasons = tosdrData.match
+
+                if (reasons.good) {
+                    reasons.good = reasons.good.map(utils.capitalizeFirstLetter)
+                }
+
+                if (reasons.bad) {
+                    reasons.bad = reasons.bad.map(utils.capitalizeFirstLetter)
+                }
```

Comment:
I'm not sure if this stuff is better done in the script that generates TOS;DR data in the first place.